src/pybind/mgr/cephadm/module.py

```diff
@@ -3055,8 +3055,27 @@ def set_prometheus_access_info(self, user: str, password: str) -> str:
         self.set_store(PrometheusService.PASS_CFG_KEY, password)
         return 'prometheus credentials updated correctly'
 
+    @handle_orch_error
+    def set_prometheus_cert(self, cert: str) -> str:
+        self.set_store(PrometheusService.PROMETHEUS_CERT_CFG_KEY, cert)
+        return 'prometheus cert stored correctly'
+
+    @handle_orch_error
+    def get_prometheus_cert(self) -> str:
+        prometheus_cert = self.get_store(PrometheusService.PROMETHEUS_CERT_CFG_KEY)
+        return prometheus_cert
+
     @handle_orch_error
     def set_prometheus_target(self, url: str) -> str:
+        if url.startswith("http://") or url.startswith("https://"):
+            return f"Invalid URL '{url}'. It should be in the format host_ip:port"
+        try:
+            host, port = url.rsplit(":", 1)
+            ipaddress.ip_address(host)
+            if int(port) < 1 or int(port) > 65535:
+                raise ValueError
```

Comment:
since we are doing an explicit check for the port here we should be explicit also about the error and added `invalid port value {port}. Valid range is 1-65535`

Reply:
Updated the PR